lib/migration-payloads/validation/index.js

```diff
@@ -33,6 +33,23 @@ const checkIfDisplayFieldsExist = function (payload) {
   return errors;
 };
 
+const checkIfNewIdIsUnique = function (payload) {
+  const errors = [];
+  const fields = payload.payload.fields;
+  const fieldsWithNewIds = fields.filter((field) => field.newId);
+  fieldsWithNewIds.forEach((field) => {
```

Comment:
Can we have a `for-of` or `.reduce()` here? List functionals with side-effects scare me 👻